modules/experimental/src/gpgpu/point-in-polygon/gpu-point-in-polygon.md

```diff
@@ -0,0 +1,81 @@
+# GPUPointInPolygon (WebGL2)
+
+`GPUPointInPolygon` provides GPU accelerated PIP (Point-In-Polygon) testing functionality. A given set of 2D points and one or more 2D polygons, it computes, whether each point is inside or outside of any polygon.
+
+## Sample Usage
+```js
+
+// construct data into required formats
+const polygons =
+  [[-0.5, -0.5], [0.5, -0.5], [0.5, 0.5], [-0.5, 0.5], [-0.5, -0.5]] // polygon vertices
+];
+
+// XY locations of 6 points
+const points = [
+  0, 0,
+  5.0, -0.25,
+  0.25, -0.25,
+  0.25, 8.25,
+  -0.25, 0.25,
+  -3.45, 0.45
+];
+
+const positionBuffer = new Buffer(gl2, new Float32Array(points));
+const count = 6;
+// Allocate result buffer with enough space (2 floats for each point)
+const filterValueIndexBuffer = new Buffer(gl2, count * 2 * 4);
```

Comment:
is there any reason why I have to create `filterValueIndexBuffer`, why can't it be associated with the `positionBuffer`? and I just call

`positionBuffer.getFilteredData();`

Also the current implementation seems completely independent of Layer. But most of the time people want to apply the polygon filters for an existing point layer. in which case the position Buffer has already existed as `instancePositions`. Would it be possible to have some kind of API to obtain the existing positionBuffer from a layer? Or better create a PolygonFilter layer extension that does this under the hood.

Reply:
The idea to seprerate them, is so users can use existing positionBuffer, if already have it.

This is luma.gl module, `Layer` support will be part of deck.gl feature either through `Effect` or `LayerFilter`.